pint_server/app.py

```diff
@@ -765,17 +774,110 @@ def assert_valid_date(date):
         abort(Response('', status=404))
 
 
-def make_response(content_dict, collection_name, element_name):
-    if request.path.endswith('.xml'):
-        return Response(
-            json_to_xml(content_dict, collection_name, element_name),
-            mimetype='application/xml;charset=utf-8')
+def supported_encoding(accept_encoding):
+    return (accept_encoding is not None) and any(comp_type in accept_encoding for comp_type in (
+            'bzip2', 'gzip', 'lzma'))
+
+
+def make_response_payload(
+        content_dict, collection_name, element_name, charset):
+    # generate xml or json formatted payload
+    if 'xml' in request.path:
+        payload = json_to_xml(content_dict, collection_name, element_name)
+        app_type = 'xml'
     else:
         if collection_name:
             content = {collection_name: content_dict}
         else:
             content = content_dict
-        return jsonify(**content)
+        payload = json.dumps(content, separators=(',', ':'))
+        app_type = 'json'
+
+    # encode the payload with the desired charset
+    payload = payload.encode(charset)
+    return app_type, payload
+
+
+def make_response(content_dict, collection_name, element_name):
+
+    charset = 'utf-8'
+    content_encoding_header = None
+    accept_encoding = None
+    app_type, payload = make_response_payload(
+        content_dict, collection_name, element_name, charset)
+    content_type = 'application/%s;charset=%s' % (app_type, charset)
+
+    if request.headers.get('Accept-Encoding') != None:
+        accept_encoding = request.headers.get('Accept-Encoding').split(", ")
```

Comment:
Here you are splitting the value of the accept_encoding header on `, ` but above on line 698 & 699 you don't split it, and pass it as the string value.

In both cases the `in` check in the comprehension argument to the any() call in supported_encoding() will work, but in this case it will only match for exact matches, without risk of substring partial matches, because you are checking for the existence of an entry in a list or tuple... However, in the earlier case it is still susceptible to substring partial matches, since it is just checking for a substring being present in a string.

Finally the standard allows for the whitespace around the comma being optional, so you may be better off doing a split on `,` and then apply a strip() to the resulting list elements:

```suggestion
        accepted_encodings = [e.strip() for e in accept_encoding.split(",")]
```